src/realm/geospatial.hpp

```diff
@@ -47,24 +48,41 @@ struct GeoPoint {
         , altitude(alt)
     {
     }
-    double longitude = 0;
-    double latitude = 0;
+
+    double longitude = get_nan();
+    double latitude = get_nan();
+    double altitude = get_nan();
+
     bool operator==(const GeoPoint& other) const
     {
-        return longitude == other.longitude && latitude == other.latitude && get_altitude() == other.get_altitude();
+        return longitude == other.longitude && latitude == other.latitude &&
+               ((!has_altitude() && !other.has_altitude()) || altitude == other.altitude);
```

Comment:
With the default constructor deleted, I think this is fairly safe. But to protect from surprising behaviour when a user sets lon/lat to NaN, what do you think about this:
```suggestion
        return (longitude == other.longitude || (std::isnan(longitude) && std::isnan(other.longitude)) 
            && (latitude == other.latitude || (std::isnan(latitude) && std::isnan(other.latitude)) 
            && ((!has_altitude() && !other.has_altitude()) || altitude == other.altitude);
```